### PR TITLE
SB Swagger Spec

### PR DESCRIPTION
**Pull request by sazeesha**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@sazeesha__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (sazeesha). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: arm-servicebus/2014-09-01/swagger/servicebus.json [hunk 1]
```diff
@@ -0,0 +1,937 @@
...
+            "EventHub"
+          ],
+          "x-ms-enum": {
+            "name": "kind",
+            "modelAsString": true
+          }
+        },
+        "sku": {
+          "$ref": "#/definitions/ARMSku",
+          "description": "Sku of the Namespace."
```
Comment (fearthecowboy) [inline arm-servicebus/2014-09-01/swagger/servicebus.json]: Sibling properties of the `$ref` should actually be in the referenced definition. (Extra JSON Reference properties will be ignored)

File: arm-servicebus/2014-09-01/swagger/servicebus.json [hunk 2]
```diff
@@ -0,0 +1,937 @@
...
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
```
Comment (fearthecowboy) [inline arm-servicebus/2014-09-01/swagger/servicebus.json]: Sibling properties of the `$ref` should actually be in the referenced definition. (Extra JSON Reference properties will be ignored)

File: arm-servicebus/2014-09-01/swagger/servicebus.json [hunk 3]
```diff
@@ -0,0 +1,937 @@
...
+        "location": {
+          "type": "string",
+          "description": "Namespace data center location."
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the Namespace AuthorizationRule."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
```
Comment (fearthecowboy) [inline arm-servicebus/2014-09-01/swagger/servicebus.json]: Sibling properties of the `$ref` should actually be in the referenced definition. (Extra JSON Reference properties will be ignored)

File: arm-servicebus/2014-09-01/swagger/servicebus.json [hunk 4]
```diff
@@ -0,0 +1,937 @@
...
+      "required": [
+        "properties"
+      ],
+      "description": "Parameters supplied to the CreateOrUpdate Namespace AuthorizationRules."
+    },
+    "SharedAccessAuthorizationRuleResource": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SharedAccessAuthorizationRuleProperties",
```
Comment (fearthecowboy) [inline arm-servicebus/2014-09-01/swagger/servicebus.json]: Sibling properties of the `$ref` should actually be in the referenced definition. (Extra JSON Reference properties will be ignored)

File: arm-servicebus/2014-09-01/swagger/servicebus.json [hunk 5]
```diff
@@ -0,0 +1,937 @@
...
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Namespace tags."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NamespaceProperties",
```
Comment (fearthecowboy) [inline arm-servicebus/2014-09-01/swagger/servicebus.json]: Sibling properties of the `$ref` should actually be in the referenced definition. (Extra JSON Reference properties will be ignored)

File: arm-servicebus/2014-09-01/swagger/servicebus.json (hunk 4 above)
Comment (sazeesha) [inline arm-servicebus/2014-09-01/swagger/servicebus.json]: Can you please elaborate what is missing here?

**Comment by fearthecowboy** [comment]
@sazeesha it's ok just to push new commits to your fork when you have an open PR, GitHub will see that and update the PR automatically.  That way we can see the changes from version-to-version, and fixes to lines will hide the line comments 😸

**Comment by fearthecowboy** [comment]
👍 
Looks Good To Me ™️ 

@amarzavery looks like everything in yours and mine comments addressed. What do you think?

File: arm-servicebus/2014-09-01/swagger/servicebus.json [hunk 6]
```diff
@@ -0,0 +1,931 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
```
Comment (amarzavery) [inline arm-servicebus/2014-09-01/swagger/servicebus.json]: @sazeesha - I thought you agreed that this was a long running operation, when I reviewed it last time? 
Is it indeed long running? If yes then please add the `"x-ms-long-running-operation": true` extension on this operation.

Comment (sazeesha) [inline arm-servicebus/2014-09-01/swagger/servicebus.json]: Yes that is correct. I have added it back

File: arm-servicebus/2014-09-01/swagger/servicebus.json [hunk 7]
```diff
@@ -0,0 +1,931 @@
...
+            "Basic",
+            "Standard",
+            "Premium"
+          ],
+          "x-ms-enum": {
+            "name": "Tier",
+            "modelAsString": true
+          }
+        },
+        "Capacity": {
```
Comment (amarzavery) [inline arm-servicebus/2014-09-01/swagger/servicebus.json]: is there a max/min constraint on this property that the server enforces? If yes, then you should be specifying that over here. Check this [link](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#patterned-fields-2). Once the page is loaded then scroll up and you will see the set of constraints that one can specify.

**Comment by amarzavery** [comment]
Otherwise the spec LGTM.

File: arm-servicebus/2014-09-01/swagger/servicebus.json (hunk 7 above)
Comment (sazeesha) [inline arm-servicebus/2014-09-01/swagger/servicebus.json]: There is no max or min, Capacity is determined by the type of the resource. If resource is EventHub namespace the capacity is 1-20 and if it is Messaging with Premium tier it is either 1,2 or 4 and if other skus it is default.

File: arm-servicebus/2014-09-01/swagger/servicebus.json [hunk 8]
```diff
@@ -0,0 +1,932 @@
...
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Namespace create request accepted",
+            "schema": {
+              "$ref": "#/definitions/NamespaceResource"
+            }
+          },
+        "x-ms-long-running-operation": true
```
Comment (amarzavery) [inline arm-servicebus/2014-09-01/swagger/servicebus.json]: This should be after the curly brace on line 125. The CI is failing
```
1) Azure Swagger Schema Validation /home/travis/build/Azure/azure-rest-api-specs/arm-servicebus/2014-09-01/swagger/servicebus.json should be valid Swagger:

328     Uncaught Error: Schema validation failed: [

329	{

330		"code": "OBJECT_ADDITIONAL_PROPERTIES",

331		"params": [

332			[

333				"x-ms-long-running-operation"

334			]

335		],

336		"message": "Additional properties not allowed: x-ms-long-running-operation",

337		"path": "#/paths/~1subscriptions~1{subscriptionId}~1resourceGroups~1{resourceGroupName}~1providers~1Microsoft.ServiceBus~1namespaces~1{namespaceName}/put/responses",

338		"schemaId": "https://raw.githubusercontent.com/Azure/autorest/master/schema/swagger-extensions.json#",

339		"description": "Response objects names can either be any valid HTTP status code or 'default'."

340	}

341]

```

**Pull request merged**